thoth-export-server/src/marc21/marc21record_thoth.rs

```diff
@@ -532,6 +553,47 @@ fn language_field(languages: &[WorkLanguages]) -> Option<FieldRepr> {
     Some(language_field)
 }
 
+fn toc_field(relations: &[WorkRelations]) -> ThothResult<FieldRepr> {
+    let mut chapters = relations
+        .iter()
+        .filter(|r| r.relation_type == RelationType::HAS_CHILD)
+        .collect::<Vec<_>>();
+    if chapters.is_empty() {
+        return Err(ThothError::IncompleteMetadataRecord(
+            MARC_ERROR.to_string(),
+            "No chapter data available for constructing contents note".to_string(),
+        ));
+    }
+    // WorkQuery should already have retrieved these sorted by ordinal, but sort again for safety
+    chapters.sort_by(|a, b| a.relation_ordinal.cmp(&b.relation_ordinal));
+    let mut toc_field: FieldRepr = FieldRepr::from((b"505", "00"));
+    let mut separator = " --";
+    for (i, chapter) in chapters.iter().enumerate() {
+        let chapter_title = &chapter.related_work.full_title;
+        let chapter_number = chapter.relation_ordinal;
+        let chapter_authors = chapter
+            .related_work
+            .contributions
+            .iter()
+            .map(|c| c.full_name.clone())
+            .collect::<Vec<_>>()
+            .join(", ");
+        toc_field = toc_field.add_subfield(b"g", format!("{}.", chapter_number))?;
```

Comment:
I'm not entirely sure we should use the ordinal as chapter number, as in some cases (e.g. a foreword) it doesn't really correspond to the number. Apparently this subfield (g) can also be used for things like page ranges, so perhaps that's a better use of it

Reply:
Yes, true. [The example given in feedback](https://cloud.obpcloud.org/f/6674) used $g for chapter numbers, but [the LOC MARC guidelines](https://www.loc.gov/marc/bibliographic/bd505.html) suggest that "chapter numbers are generally omitted"...!